src/form/Form.tsx

```diff
@@ -953,6 +962,63 @@ function Form({
     });
     if (invalid) return;
 
+    // Perform any Stripe payment method setup here if and only if the
+    // payment method field has a complete value
+    const paymentMethodFields = getPaymentMethodFieldsOnStep(activeStep);
+    const haveCompletePaymentMethod = paymentMethodFields.some((pmf: {servar: {key: string}}) =>
+      // @ts-expect-error
+      Boolean(formattedFields[pmf.servar.key].value?.complete)
+    );
+    // @ts-expect-error
+    if (integrations.stripe && haveCompletePaymentMethod) {
```

Comment:
all of this logic should live in handleActions below, which was designed to be the entrypoint for integration submissino logic

Reply:
in general we should minimize the amount of stripe-specific logic in Form.tsx, which should be for pure form logic and is already quite hefty.

Reply:
Ok, good to know regarding the purpose of handleActions.  I missed that.  I moved the stripe card setup to there which is cleaner and makes use of the existing design to set the field error, etc.

Reply:
Regarding the second comment, I moved the bulk of the code into stripe.js in Integrations in a new method: setupPaymentMethods.